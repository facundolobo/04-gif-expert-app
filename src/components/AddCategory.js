import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange= (e) =>{
        setInputValue(e.target.value);

        console.log('handleInputChange llamado')
    }

    const handleSubmit =(e) => {
        e.preventDefault();
    
        // console.log('submit hecho')
        if(inputValue.trim().length >2){
            setCategories(cats => [inputValue,...cats,]); //agrega lo del input a la lista
            setInputValue(''); //borra el input
        }
        
    }


    return (
        
            <form onSubmit={handleSubmit}>
            <p> {inputValue}</p>
            <input
            type="text"
            value={inputValue}
            onChange={handleInputChange}
            />
                
            </form>
        
    )
}

AddCategory.PropTypes = {
    setCategories: PropTypes.func.isRequired
}