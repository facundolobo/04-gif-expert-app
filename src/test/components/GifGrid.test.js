import React from 'react';
import { shallow } from "enzyme"; 
import { GifGrid } from "../../components/GifGrid"; 
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //finge cualquier llamada a ese archivo y suponer o controlar la info q eso va a responder


describe('Pruebas en <GifGrid/>', () => {
    
    const category = 'One Puch'; //nueva funcion
    
    test('debe de mostrar correctamente ', () => {
        //falseamos la data
        
        useFetchGifs.mockReturnValue ({  
            data: [],
            loading: true
        });

        const wrapper= shallow(<GifGrid category={category}/>)
        
        expect(wrapper).toMatchSnapshot();

    })
    test('debe de mostrar items cuandos e cargan imágenes useFetchGifs', () => {
        
        const gifs = [{  //lo mandamos en data 
            id: 'ABC',
            title: 'cualquier cosa',
            url: 'http://localhost.cualquiercosa.jpg'
        }]
        
        useFetchGifs.mockReturnValue ({  
            data: gifs,  //aca
            loading: false  //osea ya tengo la información
        });
        
        const wrapper= shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();

    })
    
    
})
