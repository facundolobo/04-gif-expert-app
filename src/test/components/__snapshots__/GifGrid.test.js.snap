// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de mostrar correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Puch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid/> debe de mostrar items cuandos e cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Puch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="http://localhost.cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
